device/tt_device.h

```diff
@@ -238,6 +241,15 @@ class tt_device
         throw std::runtime_error("---- tt_device::set_device_dram_address_params is not implemented\n");
     }
 
+    /**
+     * Set NOC parameters used by UMD to communicate with the TT Device (used for remote transactions).
+     *
+     * @param noc_params_ All the NOC parameters required by UMD.
+     */
+    virtual void set_driver_noc_params(const tt_driver_noc_params& noc_params_) {
```

Comment:
Isn't this hardware specific? We can't set custom noc parameters, it won't work, right? To be clear, I'd expect that noc_params are not needed at all.

Reply:
@joelsmithTT also mentioned he doesn't think we should need those parameters, but on the surface they appear used:

https://github.com/tenstorrent/tt-umd/blob/dfbd4bf648d5561e627c4b876c248cc6feae75aa/device/tt_silicon_driver.cpp#L1376-L1388

Any guidance on how to remove them?

I would love to not need this new API.

Reply:
Those parameters are used only in this `get_sys_addr` function -- as far as I can tell (?).  I think the simplest thing to do is define them as constants in `get_sys_addr`.  36 and 6.  If Blackhole and Wormhole ethernet firmwares diverge and they need to be different, we will have bigger problems to address than this method.

That the application is passing these in seems like unnecessary complication.

Reply:
@joelsmithTT thanks for the guidance!
I made the changes, and stripped out the new API.